source/common/access_log/access_log_impl.h

```diff
@@ -173,13 +183,39 @@ class ResponseFlagFilter : public Filter {
   ResponseFlagFilter(const envoy::config::filter::accesslog::v2::ResponseFlagFilter& config);
 
   // AccessLog::Filter
-  bool evaluate(const StreamInfo::StreamInfo& info,
-                const Http::HeaderMap& request_headers) override;
+  bool evaluate(const StreamInfo::StreamInfo& info, const Http::HeaderMap& request_headers,
+                const Http::HeaderMap& response_headers,
+                const Http::HeaderMap& response_trailers) override;
 
 private:
   uint64_t configured_flags_{};
 };
 
+/**
+ * Filters requests that have a response with a gRPC status. Because the gRPC protocol does not
+ * guarantee a gRPC status code, if a gRPC status code is not available, then the filter will infer
+ * infer the gRPC status code if an HTTP status code if available.
+ */
+class GrpcStatusFilter : public Filter {
+public:
+  GrpcStatusFilter(const envoy::config::filter::accesslog::v2::GrpcStatusFilter& config);
+
+  // AccessLog::Filter
+  bool evaluate(const StreamInfo::StreamInfo& info, const Http::HeaderMap& request_headers,
+                const Http::HeaderMap& response_headers,
+                const Http::HeaderMap& response_trailers) override;
+
+private:
+  std::unordered_set<Grpc::Status::GrpcStatus, Grpc::GrpcStatusHash> statuses_;
```

Comment:
We are on C++14 here so I don't think you need to specify hash explicitly?

Reply:
The CI builds weren't compiling earlier on when I didn't explicitly provide a hash. See the CI failures on 0de9b81 for more details.

Have the CI builds been changed since I change to an explicit hash?

Reply:
Hmm, try merge master and see if it still fails? On my local gcc/llvm I can build without explicit hash.

Reply:
Though it might be due to libstdc++ incapability as LLVM also failed.

Reply:
That would make more sense than compiler differences, which had been my previous thought, because it's a matter of if std::hash is implemented on enums or not.

I'll try pushing after removing the explicit hash function and see if it still fails.

Reply:
It looks like it's being compiled correctly after the merge. Tests are failing, but that's just because my `EXPECT_THROW_WITH_MESSAGE` expects an exact message, but it looks like it's comparing against JSON that can be serialized with its fields in various orders.

I'll remove the hash, remove the commented out `unordered_map` that refers to the hash, and fix the test.